tests/test_plugin.py

```diff
@@ -17,42 +17,59 @@
+        generated, _ = pytester.inline_genitems(example, f"--track-repository={temp_repo}")
+        assert [func.name for func in generated] == expected
+
+    def test_runs_correct_race_commands(self, caplog, temp_repo, run):
+        def expected_log_line(track, challenge):
+            command = (
+                f'esrally race --track="{track}" --challenge="{challenge}" '
+                f'--track-repository="{temp_repo}" --track-revision="main" '
+                '--configuration-name="pytest" --enable-assertions --kill-running-processes '
+                '--on-error="abort" --pipeline="benchmark-only" --target-hosts="127.0.0.1:19200" --test-mode'
+            )
+
+            return ("pytest_rally.rally", "INFO", f'Running command: [{command}]')
 
-        return ("pytest_rally.rally", "INFO", f'Running command: [{command}]')
-
-    challenges = [
-        "index-and-query",
-        "index-only",
-    ]
-
-    expected = [expected_log_line("test-track", challenge) for challenge in challenges]
-    res = run()
-    actual = [(r.name, r.levelname, r.message) for r in caplog.records if "esrally race" in r.message]
-    assert actual == expected
-
-def test_runs_correct_install_command(caplog, temp_repo, run):
-    expected = [
-        ("pytest_rally.elasticsearch", "DEBUG", 'Installing Elasticsearch: '
-         '[esrally install --quiet --http-port=19200 --node=rally-node --master-nodes=rally-node '
-         '--car=4gheap,trial-license,x-pack-ml,lean-watermarks --seed-hosts="127.0.0.1:19300" '
-         '--revision=current]')
-    ]
-    res = run()
-    actual = [(r.name, r.levelname, r.message) for r in caplog.records if "esrally install" in r.message]
-    assert actual == expected
+        challenges = [
+            "index-and-query",
+            "index-only",
+        ]
+
+        expected = [expected_log_line(track, challenge) for track in self.tracks for challenge in challenges]
+        res = run()
+        actual = [(r.name, r.levelname, r.message) for r in caplog.records if "esrally race" in r.message]
+        assert actual == expected
+
+    def test_runs_correct_install_command(self, caplog, temp_repo, run):
+        expected = [
+            ("pytest_rally.elasticsearch", "DEBUG", 'Installing Elasticsearch: '
+            '[esrally install --quiet --http-port=19200 --node=rally-node --master-nodes=rally-node '
+            '--car=4gheap,trial-license,x-pack-ml,lean-watermarks --seed-hosts="127.0.0.1:19300" '
+            '--revision=current]')
+        ]
+        res = run()
+        actual = [(r.name, r.levelname, r.message) for r in caplog.records if "esrally install" in r.message]
+        assert actual == expected
+
+    def test_track_filter_limits_tracks(self, pytester, example, temp_repo):
+        # Only "test-track2" should be included when filtered
+        generated, _ = pytester.inline_genitems(
+            example,
+            f"--track-repository={temp_repo}",
+            "--track-filter=test-track2"
```

Comment:
Let's add a unit test for a comma case as well. Also, can you think of a test of a marker to complete the coverage?